pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -44,12 +53,18 @@ func (c *clientCache) ClientConfigForVersion(version string) (*client.Config, er
 			}
 		}
 	}
+	if config, ok := c.configs[version]; ok {
+		return config, nil
+	}
 	// TODO: have a better config copy method
 	config := *c.defaultConfig
-	if len(version) != 0 {
-		config.Version = version
+	negotiatedVersion, err := client.NegotiateVersion(&config, version)
+	if err != nil {
+		return nil, err
 	}
+	config.Version = negotiatedVersion
 	client.SetKubernetesDefaults(&config)
```

Comment:
It seems like I'd want to set all the default except for `Version` before negotiating the version.  That way I'd get things like the default prefix and user agent.  Is this ordered like this simply because of the version defaulting?